tensorboard/backend/application_test.py

```diff
@@ -105,6 +105,46 @@ def testPluginsListing(self):
     # Plugin foo is active. Plugin bar is not.
     self.assertEqual(parsed_object, {'foo': True, 'bar': False})
 
+class TensorboardServerBaseUrlTest(tf.test.TestCase):
+  _only_use_meta_graph = False  # Server data contains only a GraphDef
+  base_url = '/test'
+  def setUp(self):
+    plugins = [
+        FakePlugin(
+            None, plugin_name='foo', is_active_value=True, routes_mapping={}),
+        FakePlugin(
+            None, plugin_name='bar', is_active_value=False, routes_mapping={}),
+    ]
+    app = application.TensorBoardWSGI(plugins, base_url=self.base_url)
+    self.server = werkzeug_test.Client(app, wrappers.BaseResponse)
+
+  def _get_json(self, path):
+    response = self.server.get(path)
+    self.assertEqual(200, response.status_code)
+    self.assertEqual('application/json', response.headers.get('Content-Type'))
+    return json.loads(response.get_data().decode('utf-8'))
+
+  def testBaseUrlRequest(self):
+    """Request a page that doesn't exist; it should 404."""
+    response = self.server.get(self.base_url)
+    self.assertEqual(404, response.status_code)
+
+  def testBaseUrlRequestNonexistentPage(self):
+    """Request a page that doesn't exist; it should 404."""
+    response = self.server.get(self.base_url + '/asdf')
+    self.assertEqual(404, response.status_code)
+
+  def testBaseUrlNonexistentPluginsListing(self):
+    """Test the format of the data/plugins_listing endpoint."""
+    response = self.server.get('/tet/data/plugins_listing')
```

Comment:
What's `tet`?

Reply:
You can run tests by using bazel:

```
bazel test tensorboard/backend:application_test
```

from the repo directory.

Reply:
test was an example of a non-existing path_prefix, it will be fixed and clarified by replacing it with non_existing_prefix